src/recovery.rs

```diff
@@ -679,12 +724,53 @@ impl Recovery {
 
     pub fn rate_check_app_limited(&mut self) {
         if self.app_limited {
-            let limited = self.delivered + self.cc.bytes_in_flight();
+            let limited = self.delivered + self.bytes_in_flight;
             self.app_limited_at_pkt = if limited > 0 { limited } else { 1 };
         }
     }
 }
 
+/// Available congestion control algorithms.
+///
+/// This enum provides currently available list of congestion control
+/// algorithms.
+#[derive(Debug, Copy, Clone, PartialEq)]
+pub enum CongestionControlAlgorithm {
+    /// Reno congestion control algorithm (default). `reno` in a string form.
+    Reno = 0,
+}
+
+impl FromStr for CongestionControlAlgorithm {
+    type Err = crate::Error;
+
+    /// Converts a string to `CongestionControlAlgorithm`.
+    ///
+    /// If `name` is not valid, `Error::CongestionControl` is returned.
+    fn from_str(name: &str) -> std::result::Result<Self, Self::Err> {
+        match name {
+            "reno" => Ok(CongestionControlAlgorithm::Reno),
+
+            _ => Err(crate::Error::CongestionControl),
+        }
+    }
+}
+
+pub struct CongestionControlOps {
+    pub on_packet_sent: fn(r: &mut Recovery, sent_bytes: usize, now: Instant),
```

Comment:
I think we can move `now` to a field of Recovery and remove `now` from all the hooks. `now` timestamp mostly coming from the caller of Recovery (e.g. `on_packet_sent()` called in `lib.rs`) and passed to most of calls inside, so it will be simpler if we update `recovery.now` only and all functions inside can use that value without additional parameters in those calls. Similar idea to how you removed `trace_id` I think.

Reply:
`now` is not a property of `Recovery` so it doesn't belong there. Also, I think it would make it harder to see where the value is initialized (currently you just have to follow the call stack).

Reply:
My question is why not? However when I look the usage of `now` across the code, this may not be simple. Let's move on.